nrespipe/tasks.py

```diff
@@ -55,43 +55,56 @@ def run_idl(idl_procedure, args, data_reduction_root, site, nres_instrument):
             lines_to_upload = f.read().splitlines()
         for line_to_upload in lines_to_upload:
             file_to_upload, dayobs = line_to_upload.split()
-            final_product = copy_to_final_directory(file_to_upload, data_reduction_root, site, nres_instrument, dayobs)
-            post_to_fits_exchange(settings.broker_url, final_product)
-        os.remove(file_upload_list)
+            ingest_file(file_path=file_to_upload)
+    os.remove(file_upload_list)
     return console_output.returncode
 
 
 @app.task(max_retries=3, default_retry_delay=3 * 60)
 @metric_timer('nrespipe', async=False)
-def process_nres_file(path, data_reduction_root_path, db_address):
-    input_filename = os.path.basename(path)
+def process_nres_file(file_info, data_reduction_root_path, db_address):
 
-    if not os.path.exists(path):
-        logger.error('File not found', extra={'tags': {'filename': input_filename}})
-        raise FileNotFoundError
+    # If the file_info is just a string, assume it is a full path to a file
+    path = file_info.get('path')
+    if path is not None:
+        filename = os.path.basename(path)
 
-    if filename_is_blacklisted(path):
-        logger.debug('Filename does not pass black list. Skipping...', extra={'tags': {'filename': input_filename}})
+        if not os.path.exists(path):
+            logger.error('File not found', extra={'tags': {'filename': filename}})
+            raise FileNotFoundError
+
+        checksum = get_md5(path)
+    else:
+        filename = file_info.get('filename')
+        checksum = file_info.get('version_set')[0].get('md5')
+        path = None
+        if not is_raw_nres_file(file_info):
+            dbs.set_file_as_processed(filename, checksum, file_info.get('frameid'), db_address)
+
+    if filename_is_blacklisted(filename):
+        logger.debug('Filename does not pass black list. Skipping...', extra={'tags': {'filename': filename}})
         return
 
-    checksum = get_md5(path)
-    if not need_to_process(input_filename, checksum, db_address):
-        logger.debug('NRES File already processed. Skipping...', extra={'tags': {'filename': input_filename}})
+    if not need_to_process(filename, checksum, db_address):
+        logger.debug('NRES File already processed. Skipping...', extra={'tags': {'filename': filename}})
         return
 
     with tempfile.TemporaryDirectory() as temp_directory:
+        if path is None:
+            path = download_from_s3(file_info.get('frameid'), temp_directory)
```

Comment:
Looks like `download_from_s3` doesn't return anything. Should it return a path to the tempfile?